src/goose/toolkit/web_browser.py

```diff
@@ -0,0 +1,461 @@
+    #     return {}
+
+
+def get_default_browser_windows():
+    try:
+        with winreg.OpenKey(
+            winreg.HKEY_CURRENT_USER,
+            r"Software\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice"
+        ) as key:
+            prog_id, _ = winreg.QueryValueEx(key, 'ProgId')
+
+        with winreg.OpenKey(winreg.HKEY_CLASSES_ROOT, f"{prog_id}\\shell\\open\\command") as cmd_key:
+            command, _ = winreg.QueryValueEx(cmd_key, None)
+
+        if command.startswith('"'):
+            executable = command.split('"')[1]
+        else:
+            executable = command.split(' ')[0]
+
+        return os.path.basename(executable)
+
+    except Exception as e:
+        print(f"Error retrieving default browser on Windows: {e}")
+        return None
+
+def get_default_browser_macos():
+    try:
+        try:
+            result = subprocess.run(
+                "defaults read ~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers -array | grep -B 1 -A 1 http | grep LSHandlerRoleAll | awk -F= '{print $2}' | tr -d ' ;\"'",
+                shell=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True
+            )
+
+            if result.returncode == 0:
+                return result.stdout.strip()
+            else:
+                print(f"Error: {result.stderr.strip()}")
+                return None
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to get default browser on macOS: {e}")
+            return None
+    except Exception as e:
+        print(f"Error retrieving default browser on macOS: {e}")
+        return None
+
+def get_default_browser_linux():
+    try:
+        result = subprocess.run(
+            ['xdg-settings', 'get', 'default-web-browser'],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True
+        )
+
+        if result.returncode != 0:
+            print(f"Error: {result.stderr.strip()}")
+            return None
+
+        desktop_file = result.stdout.strip()
+        desktop_paths = [
+            os.path.expanduser('~/.local/share/applications/'),
+            '/usr/share/applications/',
+            '/usr/local/share/applications/',
+        ]
+
+        for path in desktop_paths:
+            desktop_file_path = os.path.join(path, desktop_file)
+            if os.path.exists(desktop_file_path):
+                with open(desktop_file_path, 'r') as f:
+                    for line in f:
+                        if line.startswith('Name='):
+                            name = line.split('=', 1)[1].strip()
+                            return name
+        return desktop_file.replace('.desktop', '')
+
+    except Exception as e:
+        print(f"Error retrieving default browser on Linux: {e}")
+        return None
+
+def get_default_browser():
```

Comment:
why did we need to do this instead of using the `webbrowser` package?

Reply:
because I am getting a name for selenium to use. I am not using the browser, rather finding which one the user prefers

Reply:
i see. so, we seem to not have any test coverage for platforms here so i'm pretty hesitant of this as a solution. 

how did you verify linux and windows? i would rather test the things we do know then add some solutions that aren't vetted.

Reply:
I have not to be honest. I was not sure if ew are ok releasing something that is mac-only